UnicornEngine/video/source/Material.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+* Copyright (C) 2017 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+
+#include <unicorn/video/Material.hpp>
+
+namespace unicorn
+{
+namespace video
+{
+Material::Material() : m_color(Color::Red)
+                     , m_isColored(true)
+                     , m_isWired(false)
+                     , m_isAlbedoExist(false)
+                     , m_albedo(nullptr)
+{
+}
+
+void Material::SetColor(glm::vec3 const& color)
+{
+    m_color = color;
+}
+
+void Material::SetAlbedo(Texture& albedo)
+{
+    m_albedo = &albedo;
+    m_isColored = false;
+    m_isAlbedoExist = true;
+}
+
+void Material::SetIsWired(bool wireframe)
+{
+    m_isWired = wireframe;
+}
+
+void Material::SetIsColored(bool colored)
+{
+    m_isColored = colored;
+}
+
+void Material::RemoveAlbedo()
```

Comment:
Can be solved via `SetAlbedo(nullptr)` after rewriting it to use pointer

Reply:
Done.